### PR TITLE
Added Rpecs and model validations

### PR DESCRIPTION
**Pull request by rebel216**
Added rpecs and model validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    it 'PostCounter must be greater than or equal to zero' do
+      @user.post_counter = -1
+      expect(@user).to_not be_valid
+    end
+
+    it 'PostCounter must be greater than or equal to zero' do
+      @user.post_counter = 5
+      expect(@user).to be_valid
+    end
```
Comment (BertrandConxy) [inline spec/models/user_spec.rb:24]: - [ ] Good job so far 👏 From the project guidelines, It is suggested to write `validation` tests together with other tests. In order to understand it better, let me give you an example for this `user_spec` 👇
```ruby 
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```
You can add tests for other validations that you added in the model 😃 👍

Comment (BertrandConxy) [inline spec/models/user_spec.rb:24]: - [ ] Also It is suggested to write unit tests for the custom methods in your models. For example in your User model, there are `recent_posts` method, please add tests for it here 😃 👍

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    it 'if comments counter greater than or equal to zero.' do
+      @post.comments_counter = -5
+      expect(@post).to_not be_valid
+    end
+
+    it 'if comments counter is integer' do
+      @post.comments_counter = 8
+      expect(@post).to be_valid
+    end
```
Comment (BertrandConxy) [inline spec/models/post_spec.rb:40]: - [ ] Same here in your `post_spec` remember the model validations you added? It is suggested to test them here also. I have given you the example in the above comments 👍

File: app/models/post.rb [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+  after_save :update_posts_counter
 
-  def update_counter
+  def update_posts_counter
     author.increment!(:post_counter)
   end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```
Comment (BertrandConxy) [inline app/models/post.rb]: - [ ] Please add tests for this custom method here 😃 👍 this helps you to discover any error that can occur in it later 😃

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @rebel216, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR

## Required Changes ♻️

- [ ] Please use the Microverse provided [README template](https://github.com/microverseinc/readme-template) and customize it to meet your project needs. Remember that this is an updated version, you are expected to use it and add content in all the sections there. 👍 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @rebel216 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Used GitFlow correctly

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Just like the previous reviewer suggested, kindly make your README professional by using the Microverse provided [README template](https://github.com/microverseinc/readme-template) while customizing it to meet your project needs 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
Hi @rebel216, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- The Pull Request has a proper title and description. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

 ## Required Changes ♻️

- [ ] Your Readme file is not descriptive. Just like the previous reviewer suggested, kindly make your README professional by using the Microverse provided [README template](https://github.com/microverseinc/readme-template) while customizing it to meet your project needs. 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider: 

- [ ] [OPTIONAL] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rebel216** [comment]
> Hi @rebel216,
> 
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights
> * The Pull Request has a proper title and description. ✔️
> * There is no linter error is present. ✔️
> * Project has met all the technical aspects you needed for this project. Keep rocking! ✔️
> 
> ## Required Changes ♻️
> * [ ]  Your Readme file is not descriptive. Just like the previous reviewer suggested, kindly make your README professional by using the Microverse provided [README template](https://github.com/microverseinc/readme-template) while customizing it to meet your project needs.
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> You can also consider:
> 
> * [ ]  [OPTIONAL] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. ❌
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Project is in middle stages , will complete the template as project goes on.Have used required template of readme file as provided by the Microverse and requested by earlier reviewers.
It has consumed all my reviews, please add these in optional suggestion.
Thank you very much.

**Review by nicupop729 — COMMENTED** [review]
Hello @rebel216  👋,

_This is an additional review as a result of the second opinion request._

Someone would say that a student in the last module would already know that a missing or incomplete README file is not acceptable here at Microverse.
We start building the README from the first PR and we update it with relevant information after each and every milestone that brings new features to the project.
I noticed that you already updated the README in the `developer` branch. 
This update was supposed to happen in this branch, the `Feature-SpecsAndValidations` branch.

It's also true that the first reviewer, @BertrandConxy was supposed to straight away invalidate the PR for the missing README. Only because of this aspect, we will approve your PR manually.

Please note that since we will approve it manually, this could take a few hours!

Keep rocking 🚀!

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @rebel216  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewers 👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**